core/src/main/java/com/linecorp/armeria/common/encoding/AbstractStreamDecoder.java

```diff
@@ -54,6 +55,10 @@ public HttpData decode(HttpData obj) {
                                               .cause(ex)
                                               .build();
             }
+            throw new UnexpectedDecodeException(
+                    "Unexpected exception has occurred. " +
+                    "This exception is not caused by the buffer reaching max size. Exception message: " +
+                    message);
```

Comment:
If I understood @ikhoon's comment correctly, you need this change, @Be-poz:
```suggestion
            } else {
                throw ex;
            }
```
and then we can remove `UnexpectedDecoderException` completely 😄